app/common/data/models_user.py

```diff
@@ -82,6 +82,32 @@ class User(BaseModel):
         order_by="Grant.name",
     )
 
+    # this has some overlap with access_grants above but is most interested in the organisation you
+    # have access to
+    _grant_recipients: Mapped[list["GrantRecipient"]] = relationship(
+        "GrantRecipient",
+        secondary="""join(
+            UserRole,
+            Organisation,
+            UserRole.organisation_id == Organisation.id
+        )""",
+        primaryjoin="User.id == UserRole.user_id",
+        secondaryjoin="""and_(
+            GrantRecipient.organisation_id == UserRole.organisation_id,
+            or_(
+                GrantRecipient.grant_id == UserRole.grant_id,
+                UserRole.grant_id.is_(None)
+            ),
+            Organisation.can_manage_grants == False
+        )""",
+        viewonly=True,
+    )
+
+    def grant_recipients(self, *, limit_to_organisation_id: uuid.UUID | None = None) -> list["GrantRecipient"]:
```

Comment:
Should this be something like `get_grant_recipients` so that it ready like something you'll call? and optionally provide params to